src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStep.java

```diff
@@ -67,16 +68,29 @@
 
     private static final long serialVersionUID = 1L;
 
-    private final @CheckForNull String label;
+    private @CheckForNull String label;
+    private int weight = 1;
 
     @DataBoundConstructor public ExecutorStep(String label) {
         this.label = Util.fixEmptyAndTrim(label);
     }
-    
+
+    @DataBoundSetter public void setLabel(String label) {
+        this.label = Util.fixEmptyAndTrim(label);
+    }
+
```

Comment:
Revert, you cannot mix `@DataBoundConstructor` and `@DataBoundSetter`.

```suggestion
```

If you want to set a weight but _not_ a label, you will need to

```groovy
node(label: null, weight: 2)
```

But in practice you would almost certainly want both.

Reply:
Would you please explain your point a bit more? Why I can't use `@DataBoundConstructor` and `@DataBoundSetter` at the same time? My idea is to preserve current behavior (`node('myLabel')`) but add an ability to set label with weight: `node(label: 'myLabel', weight: 2)` and weight only: `node(weight: 2)`.

Reply:
Any given property is either mandatory (DBC) or optional (DBS). In the case of a single mandatory property, there is a special syntax whereby the name may be omitted, hence `node('label') {…}`.

Reply:
thanks!